src/cpp/src/visual_language/pipeline.cpp

```diff
@@ -182,8 +182,16 @@ class VLMPipeline::VLMPipelineImpl : public VLMPipelineBase{
 
         m_inputs_embedder->set_apply_chat_template_status(generation_config.apply_chat_template);
 
+        // copy inputs to avoid changes in user data
+        std::vector<ov::Tensor> images;
+        for (auto rgb : rgbs) {
+            auto image = ov::Tensor{rgb.get_element_type(), rgb.get_shape()};
```

Comment:
maybe we can create Tensors on the same data?
Below:
https://github.com/openvinotoolkit/openvino.genai/blob/50c45f0a2cfefad1c7dfe85a0532abbe97525855/src/cpp/src/visual_language/inputs_embedder.cpp#L150
we don't change data, we change a way how we view it

Reply:
or maybe simplest way is to create ROI tensor where ROI = whole image size, then reshape this ROI.

ROI tensor will hold original one:
https://github.com/openvinotoolkit/openvino/blob/05bdb54f1157cd0366f3fbaea72bfd2c2ce2d40e/src/inference/src/dev/make_tensor.cpp#L291-L324

Reply:
> By the way, is it reasonable to request ov to preserve Tensor's constness? A const tensor shouldn't allow changing its shape and data() should only return a pointer to const data.

you can create such request to consider by Core team. 
but need to think about BW compatibility..

Reply:
Ticket 163832

Reply:
@sbalandi please, memory copy.

Reply:
@ilya-lavrenov yes, thank you, I am working on update, have a little problem with changing shape of ROI tensor here https://github.com/openvinotoolkit/openvino.genai/blob/master/src/cpp/src/visual_language/inputs_embedder.cpp#L159 , `Check 'new_shape.size() == m_shape.size()' failed at src/inference/src/dev/make_tensor.cpp:306` ,  trying to figure it out

Reply:
it's look as valid error, the size hasn't actually changed, just the shape

Reply:
Ticket 163965. I'll close with PR for now.